libs/blog/shared/feature-seo/src/lib/seo/seo.service.ts

```diff
@@ -0,0 +1,219 @@
+    }
+
+    if (seoData.twitter_misc) {
+      this.setMetaTwitterMisc(seoData.twitter_misc);
+    }
+  }
+
+  private setMetaTwitterMisc(miscData: object): void {
+    const entries = Object.entries(miscData);
+
+    for (const [index, entry] of entries.entries()) {
+      const label: MetaDefinition = {
+        name: `${SEO_META_KEYS.twitterMiscLabel}${index + 1}`,
+        content: entry[0],
+      };
+
+      const data: MetaDefinition = {
+        name: `${SEO_META_KEYS.twitterMiscData}${index + 1}`,
+        content: entry[1],
+      };
+
+      this._meta.updateTag(label);
+      this._meta.updateTag(data);
+    }
+  }
+
+  private setMetaDescription(metaDescription: string): void {
+    this._meta.removeTag('itemprop="description"'); // Because if we not remove the tag it will not be updated.
+
+    const keys: SeoMetaKeys[] = [
+      'description',
+      'ogDescription',
+      'twitterDescription',
+    ];
+
+    for (const key of keys) {
+      this.updateTag(metaDescription, key);
+    }
+  }
+
+  private setMetaImage(
+    images: {
+      url: string;
+      width: number;
+      height: number;
+    }[],
+  ): void {
+    for (const image of images) {
+      this._meta.removeTag('itemprop="image"'); // Because if we not remove the tag it will not be updated.
+
+      this.updateTag(image.url, 'ogImage');
+      this.updateTag(image.url, 'twitterImage');
+      this.updateTag(image.url, 'image');
+      this.updateTag(`${image.width}`, 'ogImageWidth');
+      this.updateTag(`${image.height}`, 'ogImageHeight');
+    }
+  }
+
+  private updateTag(content: string, name: SeoMetaKeys | SeoTitleKeys): void {
+    const meta: MetaDefinition = {
+      name: name,
+      content: content,
+    };
+
+    this._meta.updateTag(meta);
+  }
+
+  private removeMeta(): void {
+    Object.values(SEO_META_KEYS).forEach((key) => {
+      if (
+        key === SEO_META_KEYS.twitterMiscData ||
+        key === SEO_META_KEYS.twitterMiscLabel
+      ) {
+        // twitter:data1, twitter:data2, twitter:label1 and twitter:label2 hack
+        this._meta.removeTag(`name="${key}1"`);
+        this._meta.removeTag(`name="${key}2"`);
+      } else {
+        this._meta.removeTag(`name="${key}"`);
+        this._meta.removeTag(`itemprop="${key}"`);
+        this._meta.removeTag(`property="${key}"`);
+      }
+    });
+  }
+
+  private subscribe(): void {
+    const routerNavigation$ = this._router.events.pipe(
+      filter((event) => event instanceof NavigationEnd),
+      map(() => this._activatedRoute),
+      map((route) => {
+        while (route.firstChild) {
+          route = route.firstChild;
+        }
+        return route;
+      }),
+    );
+
+    const params$ = routerNavigation$.pipe(
+      switchMap((route) => route?.paramMap),
+    );
+    const routerData$ = routerNavigation$.pipe(
+      switchMap((route) => route?.data),
+    );
+
+    combineLatest({
+      article: toObservable(this._articleDetailsStore.articleDetails),
```

Comment:
We should avoid this. seo service is supposed to be a shared library that has no domain-related dependencies. Consider exposing methods that would be used inside the `ArticleDetailsStore`

Reply:
@DDonochVA 
Look in comment above.